context_interp/model.py

```diff
@@ -43,7 +43,7 @@ def get_forward(self, image_a, image_b, t, reuse_variables=tf.AUTO_REUSE):
 
             # Get a->b and b->a flows from PWCNet.
             # TODO: Migrate to pwcnet.get_bidirectional.
-            all_flows, _ = self.pwcnet.get_forward(from_frames, to_frames)
+            all_flows, _ = self.pwcnet.get_forward(from_frames, to_frames, reuse_variables=reuse_variables)
```

Comment:
Hmm if reuse_variables is false for whatever reason, this would cause the second set of pwcnet weights to be named differently and non-restorable? I don't know what to do here, but this seems dangerous.

Reply:
Should I force the second one to always reuse_variables = True? Would there still be this issue after migrating to pwcnet.get_bidirectional?

Reply:
Actually I am confused. So if reuse_variables is false and you want it to work, you would have to give a different name to the network right? But I still have the enclosed pwcnet with the name "pwcnet". So both networks should be able to load their pwcnet weights (restore_from just operates under different scope_prefixes).

Reply:
Actually, code crashes if get_variable is used in a reuse=False scope.
This code works if reuse=tf.AUTO_REUSE, but crashes if reuse=False.
```
import tensorflow as tf


def get_model(reuse):
    with tf.variable_scope('outer', reuse=reuse):
        with tf.variable_scope('inner', reuse=reuse):
            inputs = tf.placeholder(shape=(1, 1, 1, 1), dtype=tf.float32)
            tf.layers.conv2d(inputs=inputs, filters=1, kernel_size=[1, 1], name='conv')

          
def print_vars_after_get_model(reuse):
    get_model(reuse)
    trainable_vars = tf.trainable_variables()
    for var in trainable_vars:
        print(var.name)
            
            
reuse = tf.AUTO_REUSE
print_vars_after_get_model(reuse)
print_vars_after_get_model(reuse)
```

Reply:
reuse_variables = False is a guard against unintentional sharing, so that makes sense right?